Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/Concepts.java

```diff
@@ -201,38 +265,74 @@ private JsonValidationResult checkJsonObject(JSONObject jsonObject) {
-                JsonFields.SYNONYM_IDS).toString() : "");
-        conceptEntry.setWord(jsonObject.get(JsonFields.WORD).toString());
+        if (jsonObject.get(JsonFields.WORDNET_ID) != null) {
+            conceptEntry.setWordnetId(jsonObject.get(JsonFields.WORDNET_ID).toString());
+            String[] wrappers = jsonObject.get(JsonFields.WORDNET_ID).toString().split(Constants.CONCEPT_SEPARATOR);
+            if (wrappers.length > 0) {
+                ConceptEntry existingConceptEntry = conceptManager.getConceptEntry(wrappers[0]);
+                conceptEntry.setWord(existingConceptEntry.getWord().replace("_", " "));
+                // In case user has entered a POS value. Validate whether POS is
+                // same as wordnet POS
+                if (jsonObject.get(JsonFields.POS) != null) {
+                    if (!existingConceptEntry.getPos().equalsIgnoreCase(jsonObject.get(JsonFields.POS).toString())) {
+                        throw new POSMismatchException("Entered POS " + jsonObject.get(JsonFields.POS).toString()
+                                + " is not matching with wordnet POS " + existingConceptEntry.getPos()
+                                + ". Please enter POS matching wordnet concept POS.");
+                    }
+                }
+                conceptEntry.setPos(existingConceptEntry.getPos());
+            }
+        } else {
+            conceptEntry.setWord(jsonObject.get(JsonFields.WORD).toString());
+            conceptEntry.setPos(jsonObject.get(JsonFields.POS).toString());
+        }
         conceptEntry.setConceptList(jsonObject.get(JsonFields.CONCEPT_LIST).toString());
-        conceptEntry.setPos(jsonObject.get(JsonFields.POS).toString());
+        conceptEntry.setCreatorId(username);
+        conceptEntry.setSynonymIds(jsonObject.get(JsonFields.SYNONYM_IDS) != null
+                ? jsonObject.get(JsonFields.SYNONYM_IDS).toString() : "");
         conceptEntry.setDescription(jsonObject.get(JsonFields.DESCRIPTION).toString());
-        conceptEntry.setEqualTo(jsonObject.get(JsonFields.EQUALS) != null ? jsonObject.get(JsonFields.EQUALS)
-                .toString() : "");
-        conceptEntry.setSimilarTo(jsonObject.get(JsonFields.SIMILAR) != null ? jsonObject.get(JsonFields.SIMILAR)
-                .toString() : "");
+        conceptEntry.setEqualTo(
+                jsonObject.get(JsonFields.EQUALS) != null ? jsonObject.get(JsonFields.EQUALS).toString() : "");
+        conceptEntry.setSimilarTo(
+                jsonObject.get(JsonFields.SIMILAR) != null ? jsonObject.get(JsonFields.SIMILAR).toString() : "");
         conceptEntry.setTypeId(jsonObject.get(JsonFields.TYPE).toString());
+
         return conceptEntry;
     }
 
+    @ExceptionHandler({ POSMismatchException.class })
+    public ResponseEntity<String> posMismatchHandler(HttpServletRequest req, Exception ex) {
+        return new ResponseEntity<String>(ex.getLocalizedMessage(), HttpStatus.BAD_REQUEST);
+    }
```

Comment:
If I include the validation check in checkJsonObjectForWrapper() I can do it as in the existing